contracts/root/RootChain.sol

```diff
@@ -38,7 +38,7 @@ contract RootChain is RootChainStorage, IRootChain {
     require(dataList[4].toUint() == VOTE_TYPE, "Vote type not same");
 
     // validate hash of extradata was signed as part of the vote
-    require(keccak256(dataList[5].toBytes()) == keccak256(abi.encode(bytes20(sha256(extradata)))), "Extra data is invalid");
+    require(keccak256(dataList[5].toBytes()) == keccak256(abi.encodePacked(bytes20(sha256(extradata)))), "Extra data is invalid");
```

Comment:
Just to be sure that we are doing the same thing on other side as well, else hash won't match due to `encodePacked ` vs `encode` differences if any.

Reply:
Yes, the test passes :)

Reply:
awesome